pkg/operator2/datasync/validation_test.go

```diff
@@ -0,0 +1,351 @@
+package datasync
+
+import (
+	"fmt"
+	"testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/errors"
+	corev1listers "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func Test_validateSecret(t *testing.T) {
```

Comment:
(No action required) Consider validating the field-agnostic wrapper separately from the actual validation functions to simplify writing the tests (i.e. no indexer/lister required) and make life easier for future maintainers when they attempt to determine coverage.